identity-wallet/src/state/credentials/reducers/send_credential_request.rs

```diff
@@ -271,3 +282,144 @@ pub async fn send_credential_request(state: AppState, action: Action) -> Result<
 
     Ok(state)
 }
+
+/// Helper function to get the display name of a credential.
+fn get_credential_display_name(
+    credential_configurations_supported: &HashMap<String, CredentialConfigurationsSupportedObject>,
+    credential_configuration_id: &str,
+    verifiable_credential_record: &VerifiableCredentialRecord,
+) -> String {
+    credential_configurations_supported
+        .get(credential_configuration_id)
+        .and_then(|credential_configuration| credential_configuration.display.first())
+        // Get the name of the credential from the display property if it exists.
+        .and_then(|display| display["name"].as_str())
+        .or_else(|| {
+            // Else, if the `type` property is a string, use it as the name of the credential.
+            verifiable_credential_record.display_credential.data["type"]
+                .as_str()
+                .or(
+                    // Or, if the `type` property is an array, use the last element as the name of the credential.
+                    verifiable_credential_record.display_credential.data["type"]
+                        .as_array()
+                        .and_then(|types| types.last())
+                        .and_then(|last_type| last_type.as_str()),
+                )
+        })
+        .map(ToString::to_string)
+        // An empty string is used as a fallback. This can only happen if the credential is not a valid W3C
```

Comment:
I think we should not fallback to an empty string as it might be confusing in the frontend. Let's use some value such as "Unknown" or "Credential".

Reply:
Fixed :)